common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -199,11 +206,24 @@ protected void serviceInit(Configuration externalConf) throws Exception {
     _conf = new Configuration(externalConf);
     URL confOverlayUrl = Thread.currentThread().getContextClassLoader()
         .getResource(SHUFFLE_SERVICE_CONF_OVERLAY_RESOURCE_NAME);
+    String className = YarnShuffleService.class.getName();
     if (confOverlayUrl != null) {
-      logger.info("Initializing Spark YARN shuffle service with configuration overlay from {}",
-          confOverlayUrl);
       _conf.addResource(confOverlayUrl);
     }
+    String logsNamespace = _conf.get(SPARK_SHUFFLE_SERVICE_LOGS_NAMESPACE_KEY, "");
+    String loggerName = logsNamespace.isEmpty() ? className : className + "." + logsNamespace;
+    logger = LoggerFactory.getLogger(loggerName);
```

Comment:
Initialize `logger` to `staticLogger` and only create a custom logger in case `logsNamespace != ""` ?

```suggestion
    String logsNamespace = _conf.get(SPARK_SHUFFLE_SERVICE_LOGS_NAMESPACE_KEY, "");
    if (!logsNamespace.isEmpty()) {
      String loggerName = className + "." + logsNamespace;
      logger = LoggerFactory.getLogger(loggerName);
    }
```

Reply:
Thanks, this is much cleaner.